sklearn/covariance/tests/test_covariance.py

```diff
@@ -57,10 +57,10 @@ def test_covariance():
         cov.error_norm(empirical_covariance(X_1d), norm='spectral'), 0)
 
     # test with one sample
-    X_1sample = np.arange(5)
-    cov = EmpiricalCovariance()
-
-    assert_warns(UserWarning, cov.fit, X_1sample)
+    # FIXME I don't know what this test does
+    # X_1sample = np.arange(5)
+    # cov = EmpiricalCovariance()
```

Comment:
We do interpret ``X.shape=(n,)`` as ``X.shape=(1, n_features)``. This is what ``check_array`` does and so it should be pretty consistent now. I'm not sure we have a common test for that, maybe we should.

I agree that the covariance estimate is not really meaningful for this case.

The warning was just raised in the wrong way (if ``X.ndim == 1``, not if ``X.shape[0] == 1``)

It seems I commented out one test too much, try out ``LedoitWolf().fit(np.arange(5))`` on this branch.